packages/zoltan2/test/core/color/coloring1.cpp

```diff
@@ -105,6 +107,112 @@ int validateColoring(RCP<SparseMatrix> A, int *color)
   return nconflicts;
 }
 
+int validateDistributedColoring(RCP<SparseMatrix> A, int *color){
+  int nconflicts = 0;
+  RCP<const SparseMatrix::map_type> rowMap = A->getRowMap();
+  RCP<const SparseMatrix::map_type> colMap = A->getColMap();
+  Vector R = Vector(rowMap);
+  //put colors in the scalar entries of R
+  for(size_t i = 0; i < A->getNodeNumRows(); i++){
+    R.replaceLocalValue(i, color[i]);
+  }
+
+  Vector C = Vector(colMap);
+  Import imp = Import(rowMap, colMap);
+  C.doImport(R, imp, Tpetra::REPLACE);
+
+  Teuchos::ArrayView<const zlno_t> indices;
+  Teuchos::ArrayView<const zscalar_t> values; // not used
+
+  //count conflicts in the graph
+  //loop over local rows, treat local column indices as edges
+  size_t n = A->getNodeNumRows();
+  auto colorData = C.getData();
+  for(size_t i = 0; i < n; i++){
+    A->getLocalRowView(i, indices, values);
+    for(Teuchos_Ordinal j = 0; j < indices.size(); j++){
+      if( ((size_t)indices[j] != i) && (color[i] == colorData[indices[j]]) ){
```

Comment:
Not the right check for a diagonal entry:  need to compare global indices:
rowMap->getGlobalElement(i) != colMap->getGlobalElement(indices[j])